bundles/org.openhab.binding.homeconnect/src/main/java/org/openhab/binding/homeconnect/internal/client/HomeConnectEventSourceClient.java

```diff
@@ -0,0 +1,158 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.homeconnect.internal.client;
+
+import static org.openhab.binding.homeconnect.internal.HomeConnectBindingConstants.API_BASE_URL;
+import static org.openhab.binding.homeconnect.internal.HomeConnectBindingConstants.API_SIMULATOR_BASE_URL;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Queue;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import org.apache.commons.collections4.QueueUtils;
+import org.apache.commons.collections4.queue.CircularFifoQueue;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.homeconnect.internal.client.exception.AuthorizationException;
+import org.openhab.binding.homeconnect.internal.client.exception.CommunicationException;
+import org.openhab.binding.homeconnect.internal.client.listener.HomeConnectEventListener;
+import org.openhab.binding.homeconnect.internal.client.model.Event;
+import org.openhab.core.auth.client.oauth2.OAuthClientService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import okhttp3.Request;
+import okhttp3.sse.EventSource;
+import okhttp3.sse.EventSources;
+
+/**
+ * Server-Sent-Events client for Home Connect API.
+ *
+ * @author Jonas Brüstel - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class HomeConnectEventSourceClient {
+
+    private static final String TEXT_EVENT_STREAM = "text/event-stream";
+    private static final int SSE_REQUEST_READ_TIMEOUT = 90;
+    private static final String ACCEPT = "Accept";
+    private static final int EVENT_QUEUE_SIZE = 300;
+
+    private final String apiUrl;
+    private final EventSource.Factory eventSourceFactory;
+    private final OAuthClientService oAuthClientService;
+    private final Map<HomeConnectEventListener, EventSource> eventSourceConnections;
+    private final ScheduledExecutorService scheduler;
+    private final Queue<Event> eventQueue;
+
+    private final Logger logger;
+
+    public HomeConnectEventSourceClient(OAuthClientService oAuthClientService, boolean simulated,
+            ScheduledExecutorService scheduler, @Nullable List<Event> eventHistory) {
+        this.scheduler = scheduler;
+        this.oAuthClientService = oAuthClientService;
+
+        apiUrl = simulated ? API_SIMULATOR_BASE_URL : API_BASE_URL;
+        eventSourceFactory = EventSources.createFactory(OkHttpHelper.builder(false)
+                .readTimeout(SSE_REQUEST_READ_TIMEOUT, TimeUnit.SECONDS).retryOnConnectionFailure(true).build());
+        eventSourceConnections = new HashMap<>();
+        eventQueue = QueueUtils.synchronizedQueue(new CircularFifoQueue<>(EVENT_QUEUE_SIZE));
+        if (eventHistory != null) {
+            eventQueue.addAll(eventHistory);
+        }
+        logger = LoggerFactory.getLogger(HomeConnectEventSourceClient.class);
```

Comment:
Generally set directly when defining the variable.
Same comment for all classes.

Reply:
Is this required? I like it more the way it is. All my variable definitions are made in the constructor. I don't want to mix it.

Reply:
All other official bindings I look at in the past are probably using the other way. 
An official reviewer will let you know if this is required or not.

Reply:
Changed it.